### ISSUE TITLE
Train yolo on voc occur Resizing 544 Floating, point, exception (core, dumped)

### ISSUE THREAD

**Issue by justinshaohi** [pjreddie/darknet#140]
Hi every handsomebody，can someone help me，Follow the yolo website tutorial to train yolo，i have this problem。
environment：make darknet with cuda 8.0，opencv3.3

justin@justin-pc:~/darknet$ ./darknet detector train cfg/voc.data cfg/yolo-voc.cfg darknet19_448.conv.23
yolo-voc
layer     filters    size              input                output
    0 conv     32  3 x 3 / 1   416 x 416 x   3   ->   416 x 416 x  32
    1 max          2 x 2 / 2   416 x 416 x  32   ->   208 x 208 x  32
    2 conv     64  3 x 3 / 1   208 x 208 x  32   ->   208 x 208 x  64
    3 max          2 x 2 / 2   208 x 208 x  64   ->   104 x 104 x  64
    4 conv    128  3 x 3 / 1   104 x 104 x  64   ->   104 x 104 x 128
    5 conv     64  1 x 1 / 1   104 x 104 x 128   ->   104 x 104 x  64
    6 conv    128  3 x 3 / 1   104 x 104 x  64   ->   104 x 104 x 128
    7 max          2 x 2 / 2   104 x 104 x 128   ->    52 x  52 x 128
    8 conv    256  3 x 3 / 1    52 x  52 x 128   ->    52 x  52 x 256
    9 conv    128  1 x 1 / 1    52 x  52 x 256   ->    52 x  52 x 128
   10 conv    256  3 x 3 / 1    52 x  52 x 128   ->    52 x  52 x 256
   11 max          2 x 2 / 2    52 x  52 x 256   ->    26 x  26 x 256
   12 conv    512  3 x 3 / 1    26 x  26 x 256   ->    26 x  26 x 512
   13 conv    256  1 x 1 / 1    26 x  26 x 512   ->    26 x  26 x 256
   14 conv    512  3 x 3 / 1    26 x  26 x 256   ->    26 x  26 x 512
   15 conv    256  1 x 1 / 1    26 x  26 x 512   ->    26 x  26 x 256
   16 conv    512  3 x 3 / 1    26 x  26 x 256   ->    26 x  26 x 512
   17 max          2 x 2 / 2    26 x  26 x 512   ->    13 x  13 x 512
   18 conv   1024  3 x 3 / 1    13 x  13 x 512   ->    13 x  13 x1024
   19 conv    512  1 x 1 / 1    13 x  13 x1024   ->    13 x  13 x 512
   20 conv   1024  3 x 3 / 1    13 x  13 x 512   ->    13 x  13 x1024
   21 conv    512  1 x 1 / 1    13 x  13 x1024   ->    13 x  13 x 512
   22 conv   1024  3 x 3 / 1    13 x  13 x 512   ->    13 x  13 x1024
   23 conv   1024  3 x 3 / 1    13 x  13 x1024   ->    13 x  13 x1024
   24 conv   1024  3 x 3 / 1    13 x  13 x1024   ->    13 x  13 x1024
   25 route  16
   26 conv     64  1 x 1 / 1    26 x  26 x 512   ->    26 x  26 x  64
   27 reorg              / 2    26 x  26 x  64   ->    13 x  13 x 256
   28 route  27 24
   29 conv   1024  3 x 3 / 1    13 x  13 x1280   ->    13 x  13 x1024
   30 conv    125  1 x 1 / 1    13 x  13 x1024   ->    13 x  13 x 125
   31 detection
mask_scale: Using default '1.000000'
Loading weights from darknet19_448.conv.23...Done!
Learning Rate: 0.001, Momentum: 0.9, Decay: 0.0005
Resizing
416
段错误 (核心已转储)

**Comment by jinyu121** [comment]
(So... github just lost my answer?)

Well, I hate this problem...And have no idea since I do nt know hou to use GDB.

But I noticed that some points may cause this problem.

1. When `random=true` in cfg file, yolo will random input image size from 416 to 608. When input image size is 608, ~9G GPU memory is required. Do ou have enough GPU memory?
1. BUT many times it not crashed at the maxmium input size. I mean, it crashed many times when RESIZIG to 5xx or 4xx. I don't know why...
1. And 2 can be fix using bigger `subdivision`. for example, if `batch size=56` and `subdivision=7` crashes, you can try `subdivision=8`.
1. Another tip: If you are training a classification network using the weights provided by offcial website, you MUST do `darkne partial` thing first, even your `max_steps` is bigger than 48000.
1. I have met one situation that I can not fix using 3. That's because your network design is incorrect. you can use `random=false` when training. (BUT your net will also crash in the test parse~ Do not waist your time~)

**Comment by justinshaohi** [comment]
@jinyu121 you are so funny,thanks.
i change my opencv3.3 to opencv2.4.13,this problem does not appear again.But now ,the result ot my traing,the loss is so high,i find yolo-voc.cfg batch and subdivision should be 8 and 64, so i try traing again.

**Comment by burakekincioglu** [comment]
0 Cuda malloc failed
: File exists
darknet: ./src/utils.c:253: error: Assertion `0' failed.
Aborted (core dumped)

i also this problem,, can someone help me please

**Comment by AlexeyAB** [comment]
@burakekincioglu  If you use cuDNN - then try to use this fork: https://github.com/AlexeyAB/darknet
There problem - for some network sizes on some GPU architectures cuDNN allocates to much memory for the workspace. So for these sizes we should use algorithm without using workspace.

If you don't use cuDNN, then just increase `subdivision=16` , 32 or 64 in your cfg-file.

**Comment by cuteyuyu** [comment]
@AlexeyAB 
C:\YOLO\darknet-master\build\darknet\x64>darknet_no_gpu.exe detector train data/obj.data cfg/yolov2-tiny-obj.cfg cfg/yolov2-tiny-voc.conv.13 -dont_show
***** VIDEOINPUT LIBRARY - 0.1995 - TFW07 *****
yolov2-tiny-obj
layer     filters    size              input                output
    0 conv     16  3 x 3 / 1   416 x 416 x   3   ->   416 x 416 x  16
    1 max          2 x 2 / 2   416 x 416 x  16   ->   208 x 208 x  16
    2 conv     32  3 x 3 / 1   208 x 208 x  16   ->   208 x 208 x  32
    3 max          2 x 2 / 2   208 x 208 x  32   ->   104 x 104 x  32
    4 conv     64  3 x 3 / 1   104 x 104 x  32   ->   104 x 104 x  64
    5 max          2 x 2 / 2   104 x 104 x  64   ->    52 x  52 x  64
    6 conv    128  3 x 3 / 1    52 x  52 x  64   ->    52 x  52 x 128
    7 max          2 x 2 / 2    52 x  52 x 128   ->    26 x  26 x 128
    8 conv    256  3 x 3 / 1    26 x  26 x 128   ->    26 x  26 x 256
    9 max          2 x 2 / 2    26 x  26 x 256   ->    13 x  13 x 256
   10 conv    512  3 x 3 / 1    13 x  13 x 256   ->    13 x  13 x 512
   11 max          2 x 2 / 1    13 x  13 x 512   ->    13 x  13 x 512
   12 conv   1024  3 x 3 / 1    13 x  13 x 512   ->    13 x  13 x1024
   13 conv   1024  3 x 3 / 1    13 x  13 x1024   ->    13 x  13 x1024
   14 conv     30  1 x 1 / 1    13 x  13 x1024   ->    13 x  13 x  30
   15 detection
mask_scale: Using default '1.000000'
Loading weights from cfg/yolov2-tiny-voc.conv.13...
 seen 32
Done!
Learning Rate: 0.001, Momentum: 0.9, Decay: 0.0005
Resizing
384

and display darknet_no_gpu.exe Stop running!!!

**Comment by kalyco** [comment]
Running valgrind on `./darknet detector train data/obj.data yolo-obj.cfg darknet53.conv.74 -dont_show -mjpeg_port 8090 -map` with this issue shows
```
==19441==
==19441== Process terminating with default action of signal 8 (SIGFPE)
==19441==  Integer divide by zero at address 0x802E8270A
==19441==    at 0x41EBBB: get_random_paths(char**, int, int) (in /home/kcomalli/darknet/darknet)
==19441==    by 0x422A73: load_data_detection (in /home/kcomalli/darknet/darknet)
==19441==    by 0x423984: load_thread(void*) (in /home/kcomalli/darknet/darknet)
==19441==    by 0x50636B9: start_thread (pthread_create.c:333)
==19441==    by 0x5C2141C: clone (clone.S:109)
==19441== 
==19441== HEAP SUMMARY:
==19441==     in use at exit: 6,293,655,357 bytes in 1,300 blocks
==19441==   total heap usage: 6,961 allocs, 5,661 frees, 7,450,719,693 bytes allocated
==19441== 
==19441== Searching for pointers to 1,300 not-freed blocks
==19441== Checked 6,494,302,864 bytes
==19441== 
==19441== LEAK SUMMARY:
==19441==    definitely lost: 0 bytes in 0 blocks
==19441==    indirectly lost: 0 bytes in 0 blocks
==19441==      possibly lost: 1,824 bytes in 6 blocks
==19441==    still reachable: 6,293,653,533 bytes in 1,294 blocks
==19441==         suppressed: 0 bytes in 0 blocks
==19441== Rerun with --leak-check=full to see details of leaked memory
==19441== 
==19441== ERROR SUMMARY: 0 errors from 0 contexts (suppressed: 0 from 0)
==19441== ERROR SUMMARY: 0 errors from 0 contexts (suppressed: 0 from 0)
```
At `get_random_paths()` (line 76 in data.c) output shows that `m` was 0, causing the floating point exception on line 81: `int index = random_gen() % m;`

In `detector.c` line 112 my `m` is set to `args.m = plist->size;` which the debugger confirmed to be 0.
In my case it was incorrect pathing to my train.txt file

**Comment by YadneshHande** [comment]
tiny-yolo-voc
layer     filters    size              input                output
   0 conv     16  3 x 3 / 1   416 x 416 x   3   ->   416 x 416 x  16 0.150 BF
   1 max          2 x 2 / 2   416 x 416 x  16   ->   208 x 208 x  16 0.003 BF
   2 conv     32  3 x 3 / 1   208 x 208 x  16   ->   208 x 208 x  32 0.399 BF
   3 max          2 x 2 / 2   208 x 208 x  32   ->   104 x 104 x  32 0.001 BF
   4 conv     64  3 x 3 / 1   104 x 104 x  32   ->   104 x 104 x  64 0.399 BF
   5 max          2 x 2 / 2   104 x 104 x  64   ->    52 x  52 x  64 0.001 BF
   6 conv    128  3 x 3 / 1    52 x  52 x  64   ->    52 x  52 x 128 0.399 BF
   7 max          2 x 2 / 2    52 x  52 x 128   ->    26 x  26 x 128 0.000 BF
   8 conv    256  3 x 3 / 1    26 x  26 x 128   ->    26 x  26 x 256 0.399 BF
   9 max          2 x 2 / 2    26 x  26 x 256   ->    13 x  13 x 256 0.000 BF
  10 conv    512  3 x 3 / 1    13 x  13 x 256   ->    13 x  13 x 512 0.399 BF
  11 Cuda malloc failed
: File exists
darknet: ./src/utils.c:293: error: Assertion `0' failed.


I am trying this on google colab.When saw my RAM allocation,It look like this:
**Gen RAM Free: 12.7 GB  | Proc size: 488.1 MB
GPU RAM Free: 11370MB | Used: 71MB | Util   1% | Total 11441MB**

But it still gives me above error.I am using  **!kill -9 -1**  command to remove allocated resources.
And also used **Reset Runtime**  **Restart Runtime** multiple times.
Can someone help me to find solution.My data size is 5000 images.batch=24,subdivision=8 in my tiny-yolo-voc.cfg file

**Comment by awaisbajwaml** [comment]
> @burakekincioglu If you use cuDNN - then try to use this fork: https://github.com/AlexeyAB/darknet
> There problem - for some network sizes on some GPU architectures cuDNN allocates to much memory for the workspace. So for these sizes we should use algorithm without using workspace.
> 
> If you don't use cuDNN, then just increase `subdivision=16` , 32 or 64 in your cfg-file.

I am using cuDNN 7.6.5 and getting the following error if I make subdivision = 64


  29 conv     18       1 x 1/ 1     48 x  48 x 128 ->   48 x  48 x  18 0.011 BF
  30 yolo
[yolo] params: iou loss: mse (2), iou_norm: 0.75, cls_norm: 1.00, scale_x_y: 1.00
Total BFLOPS 6.049 
avg_outputs = 262154 
 Allocate additional workspace_size = 21.23 MB 
Loading weights from yolov3-tiny.conv.15...
 seen 64, trained: 0 K-images (0 Kilo-batches_64) 
Done! Loaded 15 layers from weights-file 
Learning Rate: 0.0008, Momentum: 0.9, Decay: 0.0005
Resizing, random_coef = 1.40 

 576 x 576 
Killed
